mne/beamformer/_lcmv.py

```diff
@@ -103,51 +138,80 @@ def _apply_lcmv(data, info, tmin, forward, noise_cov, data_cov, reg,
+                max_ori = eig_vecs[:, idx_max]            
+                Wk[:] = np.dot(max_ori, Wk)    
+                Gk = np.dot(Gk, max_ori)
+                
+                if weight_norm == 'nai':
+                    # compute spatial filter for NAI
+                    tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+                    denom = np.sqrt(noise * tmp)
+                    Wk /= denom
+                elif weight_norm == 'unit_noise_gain':
+                    # compute spatial filter for unit-noise gain
+                    tmp = np.dot(Gk.T, np.dot(Cm_inv_sq, Gk))
+                    denom = np.sqrt(tmp)
+                    Wk /= denom                                       
+                    
+                is_free_ori = False    
+                
+            else:
+                eig_vals, eig_vecs = linalg.eigh(Ck)
+
+                # Choosing the eigenvector associated with the middle 
+                # eigenvalue. The middle and not the minimal eigenvalue is used 
+                # because MEG is insensitive to one (radial) of the three dipole
+                # orientations and therefore the smallest eigenvalue reflects 
+                # mostly noise.
+                for ii in range(3):
+                    if ii != eig_vals.argmax() and ii != eig_vals.argmin():
+                        idx_middle = ii
+    
+                # TODO: The eigenvector associated with the smallest eigenvalue
+                # should probably be used when using combined EEG and MEG data
+                max_ori = eig_vecs[:, idx_middle]
+    
+                Wk[:] = np.dot(max_ori, Wk)
+                Ck = np.dot(max_ori, np.dot(Ck, max_ori))
+                is_free_ori = False
+
+                if is_free_ori:
+                    # Free source orientation
+                    Wk[:] = np.dot(linalg.pinv(Ck, 0.1), Wk)
+                else:
+                    # Fixed source orientation
+                    Wk /= Ck
 
     # Pick source orientation maximizing output source power
     if pick_ori == 'max-power':
         W = W[0::3]
 
-    # Preparing noise normalization
-    noise_norm = np.sum(W ** 2, axis=1)
-    if is_free_ori:
-        noise_norm = np.sum(np.reshape(noise_norm, (-1, 3)), axis=1)
-    noise_norm = np.sqrt(noise_norm)
+    if apply_noise_norm:
+        # Preparing noise normalization
+        noise_norm = np.sum(W ** 2, axis=1)
+        if is_free_ori:
+            noise_norm = np.sum(noise_norm.reshape(-1, 3), axis=1)
+        noise_norm = np.sqrt(noise_norm)
 
     # Pick source orientation normal to cortical surface
     if pick_ori == 'normal':
         W = W[2::3]
         is_free_ori = False
 
-    # Applying noise normalization
-    noise_norm_inv = 1. / noise_norm
-    noise_norm_inv[noise_norm == 0.] = 0.
-    if not is_free_ori:
-        W *= noise_norm_inv[:, None]
+    if apply_noise_norm:
+        # Applying noise normalization
+        noise_norm_inv = 1. / noise_norm
```

Comment:
True. I will take care of it. Maybe we also need to discuss how to deal with this option anyway - it's what has already been implemented, I just made it optional to test the new weight normalization. I guess we need to see when this option can be chosen -- not sure for example whether weight norm + noise norm really makes sense. @sarangnemo ?